### PR TITLE
Upgrade to Rust 1.70

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Upgrade our Rust version to 1.70

## Test Plan

`cargo test`

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4848** <a href="https://app.graphite.dev/github/pr/charliermarsh/ruff/4848" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/charliermarsh/ruff/4848?utm_source=stack-comment).

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-1.69
+1.70
```
Comment (charliermarsh) [inline rust-toolchain:1]: I think this is encoded in Cargo.toml too.

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
It would be nice to use the stabilized once_cell but we may still need Lazy?

File: rust-toolchain (hunk 1 above)
Comment (MichaReiser) [reply rust-toolchain:1]: Thx

**Comment by MichaReiser** [comment]
> It would be nice to use the stabilized once_cell but we may still need Lazy?

Yeah, we still need `Lazy` in many places

**Pull request merged**